hist/histpainter/src/THistPainter.cxx

```diff
@@ -7079,7 +7079,13 @@ Int_t THistPainter::PaintInit()
       }
       if (Hparam.xlowedge <=0 ) {
          if (Hoption.Same) {
-            Hparam.xlowedge = TMath::Power(10, gPad->GetUxmin());
+            TH1 *h1;
+            TIter next(gPad->GetListOfPrimitives());
+            while ((h1 = (TH1 *)next())) {
```

Comment:
When using option `same` the first histogram in the ListOfPrimitives is always the one defining the axis ?
And what is happening here if there are no histograms in the list but only other objects (e.g. TGraph's). Is it possible ?

Reply:
Yes, the first histogram encountered defines the axis. We do have this mechanism in PaintBoxes and PaintColor already when we want to paint boxes and colors relative to the previous plots. But you have a good point, this case is more general and the axis might be defined by a TGraph or something else. I will think a bit more. I would like to avoid storing the TPad limits in linear scale, just for this case. This use case appears only here.